tests/cocotb/basic/test/test_hwpe_stream_merge.py

```diff
@@ -0,0 +1,305 @@
+#---------------------------------
+# Copyright 2023 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+#---------------------------------
+
+#-----------------------------------
+# Importing useful tools 
+#-----------------------------------
+import os
+import yaml
+import random
+import math
+import sys
+
+#-----------------------------------
+# Importing cocotb 
+#-----------------------------------
+import  cocotb
+from    cocotb.triggers       import RisingEdge, Timer
+from    cocotb.clock          import Clock
+from    cocotb_test.simulator import run
+
+#-----------------------------------
+# Importing pytest
+#-----------------------------------
+import  pytest
+
+#-----------------------------------
+# Extracting and setting important variables and paths
+#-----------------------------------
+hwpe_stream_path = os.getenv("HWPE_STREAM_HOME")
+basic_path       = hwpe_stream_path + "/tests/cocotb/basic"
+
+#-----------------------------------
+# Top-level definitions 
+#-----------------------------------
+# Specify top-level module
+toplevel     = 'wrapper_hwpe_stream_merge'
+# Specify python test name that contains the @cocotb.test. Usually the name of this test.
+module       = "test_hwpe_stream_merge"
+# Specify what simulator to use (e.g., verilator, modelsim, icarus)
+simulator    = "verilator"
+# Specify build directory
+sim_build    = basic_path + "/test/sim_build/{}/".format(toplevel)
+# Get YAML source files path
+src_yml_path = hwpe_stream_path + "/src_files.yml"
+
+#-----------------------------------
+# Global parameters for testing
+# TODO: These are modifiable so change whenever needed
+#-----------------------------------
+# Checker parameter
+CHECK_COUNT   = 5
+
+# DUT parameters
+NB_IN_STREAMS = 4
+DATA_WIDTH    = 16
+
+# For random seed logging
+RANDOM_SEED = random.randrange(sys.maxsize)
+random.seed(RANDOM_SEED)
+
+
+#-----------------------------------
+# Get YAML source
+#-----------------------------------
+with open(src_yml_path,"r") as src_files:
+    try:
+        yaml_dict = yaml.safe_load(src_files)
+    except yaml.YAMLError as exc:
+        print(exc)
+
+#-----------------------------------
+# Extract source from YAML
+# Also append the main working path
+#-----------------------------------
+include_folders = yaml_dict['hwpe-stream']['incdirs']
+rtl_sources     = yaml_dict['hwpe-stream']['files']
+
+for i in range(len(rtl_sources)):
+    rtl_sources[i] = hwpe_stream_path + '/' + rtl_sources[i]
+
+for i in range(len(include_folders)):
+    include_folders[i] = hwpe_stream_path + '/' + include_folders[i]
+
```

Comment:
Is it really required to add all RTL Files to each new testbench?

Reply:
Honestly not really, but in the other directories they have a series of RTL that needs to import. We could skip it actually hmm.

Reply:
I feel like being more specific when it comes to dependencies is a good practice. Just importing everything always seems a bit brittle